rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/system_interface.kt

```diff
@@ -39,7 +41,20 @@ interface DBInterface {
 
 data class Field(val name: String, val value: Value)
 
-class Record(vararg fields: Field) : ArrayList<Field>(fields.asList())
+class Record(vararg fields: Field) : LinkedHashMap<String, Value>() {
+    init {
+        fields.forEach {
+            add(it)
+        }
+    }
+    fun matches(keyFields: List<Field>): Boolean {
+        return keyFields.all { this[it.name] == it.value }
```

Comment:
given the body is an expression you could rewrite it using the shortcut with the equals:

```
fun matches(keyFields: List<Field>) = keyFields.all { this[it.name] == it.value }
```

Reply:
Good idea